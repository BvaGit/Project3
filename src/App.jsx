import React from 'react';
import './App.scss';

const App = () => {
    return (
        <div className="b">
            Hello REact
        </div>
    )
}

export default App;