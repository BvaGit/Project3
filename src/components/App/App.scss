.b{
    font-size: 48px;
}