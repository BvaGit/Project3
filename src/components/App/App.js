import React from "react";
import { BrowserRouter as Router, Route } from 'react-router-dom';

import Auth from '../Auth';
import Registration from '../Registration';

import "./App.scss";

const App = () => {
  
  return (
    <Router>
      <Route exact path='/registration' component={Registration}/>
      <Route path='/' component={Auth}/>
    </Router>
  )
}

export default App;
