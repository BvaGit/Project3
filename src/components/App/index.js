import { connect } from 'react-redux';
import App from "./App";
import { languageValue } from '../../store/localization/selectors';

const mapStateToProps = (state) => ({
    languageValue: languageValue(state)
})

export default connect(mapStateToProps, null)(App);
