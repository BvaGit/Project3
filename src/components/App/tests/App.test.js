import React from 'react';
import { shallow, mount } from 'enzyme';
import App from '../App';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

const mockStore = configureStore();
const store = mockStore({
    auth : {
        fields: {
          login: "",
          password: "",
        },
        errors: {
          login: "",
          password: "",
        }
      },
    rooms: {
        sendingMessage: false,
        messages: {
          0: [
            {
              text: "Welcome!",
            },
          ],
        },
        socket: null,
        rooms: {},
        activeRoom: "",
      },
      user:{
        user: { id: null, token: null },
      },
      users: {
        users: [],
      },
      registration: {
        fields:{
          login: '',
          password: '',
          confirm_password:""
        },
        errors:{
          login:"",
          password:"",
          confirm_password:""
        }
      }
    });


describe('App', () => {
    it("Should match snapshot", () => {
        const component = shallowSmart(<App /> , store)
        expect(component.html()).toMatchSnapshot()
    })
})

export const shallowSmart = (component, store) => {
    const core = store
        ? <Provider store={store}>{component}</Provider>
        : component;
    return shallow(core);
};