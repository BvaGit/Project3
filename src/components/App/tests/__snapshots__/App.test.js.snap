// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should match snapshot 1`] = `"<div class=\\"gen__holder\\"><div class=\\"gen\\"><div class=\\"gen__info\\"><img src=\\"/assets/images/logo.png\\" alt=\\"logo\\" class=\\"logo\\"/><div class=\\"title\\"><h1 class=\\"title__big\\">SIGN IN</h1><br/><h2 class=\\"title__small\\">YOUR ACCOUNT</h2></div></div><div class=\\"gen__main\\"><div class=\\"gen__container\\"><label class=\\"inputs__label \\">Login<input type=\\"text\\" name=\\"login\\" value=\\"\\" title=\\"Enter your login\\" placeholder=\\"Enter your login\\" class=\\"inputs__input undefined\\" minLength=\\"1\\" maxLength=\\"30\\"/></label><label class=\\"inputs__label \\">Password<input type=\\"password\\" name=\\"password\\" value=\\"\\" title=\\"Enter your password\\" placeholder=\\"Enter your password\\" class=\\"inputs__input undefined\\" minLength=\\"1\\" maxLength=\\"30\\"/></label><div class=\\"gen__buttons\\"><button type=\\"button\\" class=\\"button\\" disabled=\\"\\">SIGN IN</button></div><p class=\\"gen__link\\">Create new account</p><a class=\\"gen__link\\" href=\\"/registration\\">Sign up</a></div></div></div></div>"`;
